hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector.java

```diff
@@ -231,22 +234,39 @@ private SlowOperationLog getOrCreate(String stackTrace, Object operation) {
 
         private void logSlowOperation(SlowOperationLog log, int totalInvocations) {
             if (isStackTraceLoggingEnabled) {
-                // log with stack traces
-                if (totalInvocations == 1) {
-                    logger.warning(format("Slow operation detected: %s%n%s", log.operation, log.stackTrace));
-                    return;
-                }
+                logWithStackTrace(log, totalInvocations);
+            } else if (!isFirstLog) {
+                logWithoutStackTrace(log, totalInvocations);
+            } else {
+                logWithConfigHint(log, totalInvocations);
+            }
+        }
+
+        private void logWithStackTrace(SlowOperationLog log, int totalInvocations) {
+            if (totalInvocations == 1) {
+                logger.warning(format("Slow operation detected: %s%n%s", log.operation, log.stackTrace));
+            } else {
                 // print the full stack trace each FULL_LOG_FREQUENCY invocations
                 logger.warning(format("Slow operation detected: %s (%d invocations)%n%s", log.operation, totalInvocations,
                         (totalInvocations % FULL_LOG_FREQUENCY == 0) ? log.stackTrace : log.shortStackTrace));
-                return;
             }
-            // log without stack traces
+        }
+
+        private void logWithoutStackTrace(SlowOperationLog log, int totalInvocations) {
             if (totalInvocations == 1) {
                 logger.warning(format("Slow operation detected: %s", log.operation));
-                return;
+            } else {
+                logger.warning(format("Slow operation detected: %s (%d invocations)", log.operation, totalInvocations));
             }
-            logger.warning(format("Slow operation detected: %s (%d invocations)", log.operation, totalInvocations));
+        }
+
+        private void logWithConfigHint(SlowOperationLog log, int totalInvocations) {
+            // print a hint once how to enable logging of stack traces
+            logger.warning(format("Slow operation detected: %s"
+                            + "%nHint: You can enable the logging of stack traces with the following config property:"
+                            + " hazelcast.slow.operation.detector.stacktrace.logging.enabled",
```

Comment:
I would use the GroupProperty for it. This way you don't run into problems when it is renamed.